nengo/connection.py

```diff
@@ -128,44 +128,47 @@ def validate(self, conn, function_info):
         transform = conn.transform
         size_mid = conn.size_in if function is None else size
 
-        if transform.ndim < 2 and size_mid != conn.size_out:
-            raise ValueError("function output size is incorrect; should "
-                             "return a vector of size %d" % conn.size_out)
+        if isinstance(transform, np.ndarray):
```

Comment:
Is it right that `DistOrArrayParam` will typecast the `transform` to an `ndarray` if it's something like a 2D-list?